src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -130,6 +135,16 @@ public void startTask(final String tableName, final TopicPartition topicPartitio
 
   @Override
   public void insert(final Collection<SinkRecord> records) {
+    Exception potentialEx = failedCleanerException.get();
+    if (potentialEx != null) {
+      for (ServiceContext pipe : pipes.values()) {
+        // flush current buffers
+        pipe.flushBuffer();
+      }
+
+      throw new ConnectException("Cleaner encountered an exception", potentialEx);
+    }
+
```

Comment:
can you extract this into a private method? may be call it something like `cleanerThreadExceptionHandling()`?